consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java

```diff
@@ -281,15 +287,83 @@ public void run() {
 
             Object[] x = shouldProcessJob(response);
             boolean processJob = (Boolean) x[0];
-            int jobExecutionPsn = (Integer) x[1];
+            jobExecutionPsn = (Integer) x[1];
 
             if (processJob) {
-                Map<String, JsonNode> sourceDocuments = getSourceDocuments();
+                currentRun.setJobStatus(JobStatus.RUNNING);
+                LightblueResponse responseMarkExecutionStatus = markExecutionStatusAndEndDate(jobExecutionPsn, currentRun.getJobStatus(), false);
+                LOGGER.debug("Updated Status Response for status {}: {}", currentRun.getJobStatus(), responseMarkExecutionStatus.getText());
 
+                Map<String, JsonNode> sourceDocuments = getSourceDocuments();
                 Map<String, JsonNode> destinationDocuments = getDestinationDocuments(sourceDocuments);
 
-                List<JsonNode> documentsToOverwrite = getDocumentsToOverwrite(sourceDocuments, destinationDocuments);
 
+                /*
+                Response can be (according from the documentation http://jewzaam.gitbooks.io/lightblue-specifications/content/language_specification/data.html ):
+                    complete: The operation completed, all requests are processed
+                    partial: Some of the operations are successful, some failed
+                    async: Operation running asynchronously
+                    error: Operation failed because of some system error
+                 */
+                boolean allComplete = true;
+                boolean anyPartial = false;
+                boolean anyAsync = false;
+                for (JsonNode jsonNode : sourceDocuments.values()) {
+                    JsonNode jsStatus = jsonNode.findValue(STATUS);
```

Comment:
Why is this looking at source document status?  The source documents are the documents returned, not the lightblue response.  If there happens to be a status element it is because that entity has a status element.